fe/fe-core/src/main/java/com/starrocks/connector/delta/ScalarOperationToDeltaLakeExpr.java

```diff
@@ -0,0 +1,384 @@
+
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import static io.delta.kernel.expressions.AlwaysTrue.ALWAYS_TRUE;
+
+public class ScalarOperationToDeltaLakeExpr {
+    public Predicate convert(List<ScalarOperator> operators, DeltaLakeContext context) {
+        DeltaLakeExprVisitor visitor = new DeltaLakeExprVisitor();
+        List<Predicate> predicates = Lists.newArrayList();
+
+        for (ScalarOperator operator : operators) {
+            Predicate predicate = operator.accept(visitor, context);
+            if (predicate != null) {
+                predicates.add(predicate);
+            }
+        }
+
+        Optional<Predicate> result = predicates.stream().reduce(And::new);
+        return result.orElse(ALWAYS_TRUE);
+    }
+
+    public static class DeltaLakeContext {
+        private final io.delta.kernel.types.StructType schema;
+        private final Set<String> partitionColumns;
+
+        public DeltaLakeContext(io.delta.kernel.types.StructType schema, Set<String> partitionColumns) {
+            this.schema = schema;
+            this.partitionColumns = partitionColumns;
+        }
+
+        boolean isPartitionColumn(String column) {
+            return partitionColumns.contains(column);
+        }
+
+        Column getColumn(String columnName) {
+            if (schema == null) {
+                return null;
+            }
+            int index = schema.indexOf(columnName);
+            if (index < 0) {
+                return null;
+            }
+            return schema.column(index);
+        }
+
+        public io.delta.kernel.types.StructType getSchema() {
+            return schema;
+        }
+    }
+
+    private static class DeltaLakeExprVisitor extends ScalarOperatorVisitor<Predicate, DeltaLakeContext> {
+        private static DeltaDataType getResultType(String columnName, DeltaLakeContext context) {
+            Preconditions.checkNotNull(context);
+            return getColumnType(columnName, context);
+        }
+
+        private static DeltaDataType getColumnType(String qualifiedName, DeltaLakeContext context) {
+            //TODO: nested type
+            StructType structType = context.getSchema();
+            StructField field = structType.get(qualifiedName);
+            if (field != null) {
+                DeltaDataType type = DeltaDataType.instanceFrom(field.getDataType().getClass());
+                // Note: A timestamp value in a partition value doesn't store the time zone due to historical reasons.
+                // It means its behavior looks similar to timestamp without time zone when it is used
+                // in a partition column.
+                // https://github.com/delta-io/delta/blob/master/PROTOCOL.md#partition-value-serialization
+                if (type == DeltaDataType.TIMESTAMP && context.isPartitionColumn(qualifiedName)) {
```

Comment:
Does the file written by deltalake have a partition column?

Reply:
If yes, do we need to consider sending partition values to be as const columns?

Reply:
No, partition column is not exist in file